File: Лабораторная работа 2/lab_python_opp/rectangle.py
from lab_python_opp.figure import Figure
from lab_python_opp.color import Color

class Rectangle(Figure):

    FIGURE_TYPE = "Прямоугольник"

    @classmethod
    def getFigureType(cls):
        return cls.FIGURE_TYPE
    
    def __init__(self, width, height, color):
        self.width = width
        self.height = height
        self.fc = Color(color)

    def area(self):
        return self.width * self.height
    
    def __repr__(self):
        return '{} {} цвета со сторонами {} и {}, площадь равна {}'.format(
            Rectangle.getFigureType(),
            self.fc.color,
            self.width,
            self.height,
            self.area()
        )

File: shapes/rectangle.cs
namespace class_lab3
{
    class Rectangle : geometricShape, IPrint
    {
        public Rectangle(double h, double w)
        {
            this.Height = h;
            this.Width = w;
        }
        public double Height { get; set; }
        public double Width { get; set; }
        public override double Area()
        {
            return Height * Width;
        }
        public override string ToString()
        {
            return ("Прямоугольник со сторонами " + Height.ToString() + " и " + Width.ToString() + "; площадь: " + this.Area());
        }
    }
}